### PR TITLE
SNOW-1617349: Add aliases for format options to dataframe reader and writer

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1617349

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   This PR adds reader and writer option aliases to help with migration. One pager about this change is here: https://docs.google.com/document/d/1U-VucApRuC74RsPEvZ5HUzqZf1NluintZ8z1Nl7fI2w/edit

### CODE REVIEW

**Review by sfc-gh-jrose — COMMENTED** [review]
The functionallity between read and write looks the same, but done in a two similar, but different ways. It might be worthwhile to extract this to a utils function to keep things dry and to make sure the error message remains consistent between the two in the future.

**Comment by sfc-gh-aalam** [comment]
> The functionallity between read and write looks the same, but done in a two similar, but different ways. It might be worthwhile to extract this to a utils function to keep things dry and to make sure the error message remains consistent between the two in the future.

Yeah. They are very similar. I added a util function to deduplicate the work

File: src/snowflake/snowpark/_internal/utils.py [hunk 1]
```diff
@@ -793,6 +793,21 @@ def get_copy_into_table_options(
     return file_format_type_options, copy_options
 
 
+def get_aliased_option_name(
+    key: str,
+    alias_map: Dict[str, str],
+) -> str:
```
Comment (sfc-gh-evandenberg) [inline src/snowflake/snowpark/_internal/utils.py:799]: Add comment?

**Review by sfc-gh-evandenberg — APPROVED** [review]
Looks good with one minor comment.

File: src/snowflake/snowpark/_internal/utils.py (hunk 1 above)
Comment (sfc-gh-aalam) [reply src/snowflake/snowpark/_internal/utils.py:799]: sure. added

**Review by sfc-gh-jrose — APPROVED** [review]

**Pull request merged**